### PR TITLE
Deprovison spec compliance

### PR DESCRIPTION
**Pull request by shawn-hurley**
Fixes issue #73

### CODE REVIEW

**Comment by jmrodri** [comment]
NACK needs work. Adding this comment in case folks don't see the review above.

**Comment by jmrodri** [comment]
These errors 

https://github.com/fusor/ansible-service-broker/blob/master/pkg/broker/broker.go#L272-L281

need to be ones we defined in this file:

https://github.com/fusor/ansible-service-broker/blob/master/pkg/broker/broker.go#L15-L19

Because as it stands now, line 272 will return an etcd error, line 281 would return something from the apb, and line 273 something different as well. So basically using the errors we defined (feel free to add a new one if necessary) return one of those as appropriate.

Then in handler.go, check the specific errors and return the correct http error. Let me know if you need more clarification.

File: pkg/broker/broker.go [hunk 1]
```diff
@@ -265,23 +276,35 @@ func (a AnsibleBroker) Deprovision(instanceUUID uuid.UUID) (*DeprovisionResponse
...
+	// bubble up  error
+	if err != nil {
+		a.log.Error("unknown etcd error - %#v", err)
 		return nil, err
 	}
-
-	if err = apb.Deprovision(instance, a.log); err != nil {
+	err = apb.Deprovision(instance, a.log)
+	// TODO: Right now the client we are using does not expose bb
+	if err == docker.ErrNoSuchImage {
```
Comment (shawn-hurley) [inline pkg/broker/broker.go]: If you follow the path of the docker client, it could come back with the image this error.

**Comment by jmrodri** [comment]
@shawn-hurley for this PR I'd like to see 2 things

1) rebase it so we no longer have conflicts
2) go to the Provision and Bind calls and change those to be consistent with your new changes. i.e. in Provision we are looking at the error string for "100".

Alternatively, change deprovision to use the same handling as provision. But I think using the clients error validation is better.

File: pkg/broker/broker.go [hunk 2]
```diff
@@ -265,23 +275,35 @@ func (a AnsibleBroker) Deprovision(instanceUUID uuid.UUID) (*DeprovisionResponse
...
-	if instance, err = a.dao.GetServiceInstance(instanceId); err != nil {
+	instance, err := a.dao.GetServiceInstance(instanceID)
+	/// Handle case where service is not found in etcd
+	if client.IsKeyNotFound(err) {
+		a.log.Debug("unable to find service instance - %#v", err)
+		return nil, ErrorNotFound
+	}
+	// bubble up  error
+	if err != nil {
+		a.log.Error("unknown etcd error - %#v", err)
```
Comment (rthallisey) [inline pkg/broker/broker.go]: Even though we aren't catching the error explicitly, I think etcd will know the error.  Maybe it should be more like ```Error from etcd - ...```.

File: pkg/broker/broker.go [hunk 3]
```diff
@@ -265,23 +275,35 @@ func (a AnsibleBroker) Deprovision(instanceUUID uuid.UUID) (*DeprovisionResponse
...
 		return nil, err
 	}
-
-	if err = apb.Deprovision(instance, a.log); err != nil {
+	err = apb.Deprovision(instance, a.log)
+	// TODO: Right now the client we are using does not expose bb
```
Comment (shawn-hurley) [inline pkg/broker/broker.go]: I can't remember, so removing.

File: pkg/broker/broker.go [hunk 4]
```diff
@@ -37,6 +43,7 @@ type AnsibleBroker struct {
 	engine        *WorkEngine
 }
 
+// NewAnsibleBroker - creates a new ansbile broker
```
Comment (jmrodri) [inline pkg/broker/broker.go]: typo: ansbile -> ansible

**Comment by rthallisey** [comment]
@shawn-hurley LGTM. Just needs a rebase.

**Pull request merged**